check_tools/find_bad_code_snippets.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -eo pipefail
+
+mkdir -p BUILD
+echo '*' > BUILD/.mbedignore
+
+# simple fake main for snippets without a main function
+cat > fake_main.c <<MAIN
+#include "mbed_toolchain.h"
```

Comment:
Is the fake_main only used to pull in the proper main.{c,cpp} file?

Reply:
The script creates a main.cpp or main.c as necessary. fake_main just provides a weak main that is a noop which gets removed at linking time if the example already contains a main.

This is a way to get examples with only class declarations to compile. Otherwise the linker would complain about a missing main function. There may be a different way to do this with compiler flags, but I was lazy.